tests/test_preprints.py

```diff
@@ -794,3 +796,49 @@ def test_creator_gets_email(self, send_mail):
 
         self.preprint_branded.set_published(True, auth=Auth(self.user), save=True)
         assert_equals(send_mail.call_count, 2)
+
+class TestWithdrawnPreprint(OsfTestCase):
+    def setUp(self):
+        super(TestWithdrawnPreprint, self).setUp()
+        self.user = AuthUserFactory()
+        self.preprint_pre_mod = PreprintFactory(provider__reviews_workflow='pre-moderation', is_published=False)
+        self.preprint_post_mod = PreprintFactory(provider__reviews_workflow='post-moderation', is_published=False)
+        self.preprint = PreprintFactory()
+
+    @mock.patch('website.preprints.tasks.get_and_set_preprint_identifiers')
+    def test_withdrawn_preprint(self, _):
+        # test_ever_public
+
+        # non-moderated
+        assert_true(self.preprint.ever_public)
```

Comment:
It looks like most of the rest of this file is not using nose-style asserts where not necessary. We're trying to move towards pytest practices, so please change this (and others below in this test) to `assert <attribute> is <boolean>`

Reply:
In past, we were converting entire test files from nose to pytest all at once. In fear it may throw people off seeing a mix of both nose and pytest. for small files, I usually convert the whole file but when it's largely nose, I leave it be.

Reply:
[There](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L284) [are](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L298) [already](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L312) [lots](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L323-L328) [of](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L344) ["raw"](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L350-L351), [non](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L358-L359)-[nose](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L404-L417) [asserts](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L425-L430) [in](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L449) [this](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L452) [file](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L469). [If](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L478-L500) [anything](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L505-L510), [nose](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L517)-[style](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L527-L536) [would](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L555-L595) [throw](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L619) [people](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L626-L655) [off](https://github.com/fabmiz/osf.io/blob/3879b5232b87f652ea9a1ef0aca5644c13beb71b/tests/test_preprints.py#L673-L762).